### PR TITLE
doc: HashiCorp external secret operator

### PR DESCRIPTION
**Pull request by Shubham9t9**
# Description

### CODE REVIEW

File: docs/user-guide/creating-application/eso/aws-eso.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 To add secrets from **AWS Secrets Manager**, we need to create a generic Kubernetes secret for AWS authentication.
 
-Create a Kubernetes secret in the namespace in which the application is to be deployed using base64 encoded AWS access-key and secret-access-key. You can use devtron generic chart for this.
+Create a Kubernetes secret in the namespace in which the application is to be deployed using base64 encoded AWS access-key and secret-access-key. You can use Devtron generic chart for this.
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/aws-eso.md:5]: Create a Kubernetes secret in the namespace in which the application is to be deployed using base64 encoded AWS access-key and secret-access-key. You can use a Devtron generic chart for to achieve it.

File: docs/user-guide/creating-application/eso/aws-eso.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
 
 **Note:** You don't have to create the Kubernetes secret everytime you create External Secret for the respective namespace.
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/aws-eso.md:7]: **Note:** You don't have to create the Kubernetes secret every time you create an external secret for the respective namespace.

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+   name: vault-token
+   namespace: <namespace>
+```
+
+**Note:** Please note that you don't need to create the Kubernetes secret every time you create an External Secret for the corresponding namespace.
+
+Once you have created the generic secret, follow these steps in the application's Secrets section:
+
+**1. Create a new secret.**
+
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:22]: **1. Create a new secret**

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+To add a new secret to the application, go to the `App Configuration` section of the application. Then, navigate to the left pane and select the `Secrets` option and click on the `Add Secret` button.
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:23]: To add a new secret to the application, go to the `App Configuration` section of the application. Then, navigate to the left pane and select the `Secrets` option and click the **Add Secret** button.

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/hc-add-secret.jpg)
+
+**2. Select `HashiCorp Vault` as the External Secret Operator**
+
+After clicking on the `Add Secret` button, select `HashiCorp Vault` from the dropdown menu for the `Data type` option. Provide a name for the secret you are creating, and then proceed to configure the external secret as described in the next step.
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:29]: After clicking the **Add Secret** button, select `HashiCorp Vault` from the dropdown menu for the `Data type` option. Provide a name for the secret you are creating, and then proceed to configure the external secret as described in the next step.

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/hc-secret-type.jpg)
+
+**3. Configure secret:**
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:33]: **3. Configure the secret**

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+| Key | Description |
+| :--- | :--- |
+| `vault.server` | Server is the connection address for the Vaultserver, e.g: "https://vault.example.com:8200". |
+| `vault.path` | Specify the path where the secret is stored in Vault. |
+| `tokenSecretRef.name` | Enter the name of the secret that will be used for authentication. |
+| `tokenSecretRef.key` | Specify the key name within the secret that contains the token. |
+| `secretKey` | Provide a name for the secret in Kubernetes. |
+| `key` | Enter the name of the secret in Vault. |
+| `property` | Specify the key within the Vault secret. |
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:48]: Remove full-stop after each one-liner description in the table

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/tokenSecretRef.jpg)
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/secrets/hc-eso.jpg)
+
+**4. Save secret.**
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:54]: **4. Save the secret**

File: docs/user-guide/creating-application/eso/hashicorp-eso.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+After configuring the external secret from HashiCorp Vault, proceed to save the secret by clicking on the `Save` button.
```
Comment (ashokdevtron) [inline docs/user-guide/creating-application/eso/hashicorp-eso.md:56]: After configuring the external secret from HashiCorp Vault, proceed to save the secret by clicking the **Save** button.

**Review by ashokdevtron — DISMISSED** [review]

File: docs/user-guide/creating-application/eso/aws-eso.md (hunk 1 above)
Comment (Shubham9t9) [reply docs/user-guide/creating-application/eso/aws-eso.md:5]: You can use a Devtron generic chart _for to_ achieve it. ?

Comment (ashokdevtron) [reply docs/user-guide/creating-application/eso/aws-eso.md:5]: @Shubham9t9, I have changed the sentence.

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**